src/Compression/CompressionCodecEncrypted.h

```diff
@@ -1,104 +1,117 @@
+    * no longer be compressible in any meaningful way. This means if
+    * you want to apply both compression and encryption to your
+    * columns, you need to put this codec at the end of the chain
+    * like "column Int32 Codec(Delta, LZ4,
+    * AES_128_GCM_SIV)".
+    *
+    * The key is obtained by executing a command specified in the
+    * configuration file at startup, and if it doesn't specify a
+    * command the codec refuses to process any data. The command is
+    * expected to write a Base64-encoded key of any length, and we
+    * apply HKDF-SHA-256 to derive a 128-bit key-generation key
+    * (only the first half of the result is used). We then encrypt
+    * blocks in AES-128-GCM-SIV with a universally fixed nonce (12
+    * repeated NUL characters).
+    *
+    * This construct has a weakness due to the nonce being fixed at
+    * all times: when the same data block is encrypted twice, the
+    * resulting ciphertext will be exactly the same. We have to live
+    * with this weakness because ciphertext must be deterministic,
+    * as otherwise our engines like ReplicatedMergeTree cannot
+    * deduplicate data blocks.
+    */
+class CompressionCodecEncrypted : public ICompressionCodec
+{
+public:
+    /** If a key is available, the server is supposed to
+        * invoke this static method at the startup. The codec will
+        * refuse to compress or decompress any data until that. The
+        * key can be an arbitrary octet string, but it is
+        * recommended that the key is at least 16 octets long.
+        *
+        * Note that the key is currently not guarded by a
+        * mutex. This method should be invoked no more than once.
+        */
+    CompressionCodecEncrypted();
+
+    uint8_t getMethodByte() const override;
+    void updateHash(SipHash & hash) const override;
+
+    bool isCompression() const override
     {
-    public:
-        /** If a master key is available, the server is supposed to
-          * invoke this static method at the startup. The codec will
-          * refuse to compress or decompress any data until that. The
-          * key can be an arbitrary octet string, but it is
-          * recommended that the key is at least 16 octets long.
-          *
-          * Note that the master key is currently not guarded by a
-          * mutex. This method should be invoked no more than once.
-          */
-        static void setMasterKey(const std::string_view & master_key);
+        return false;
+    }
 
-        CompressionCodecEncrypted(const std::string_view & cipher);
+    bool isGenericCompression() const override
+    {
+        return false;
+    }
 
-        uint8_t getMethodByte() const override;
-        void updateHash(SipHash & hash) const override;
+    bool isPostProcessing() const override
+    {
+        return true;
+    }
+    static void updateEncryptionKeys(const Poco::Util::AbstractConfiguration & config, const String & config_prefix);
```

Comment:
I don't think we need to have this function. Just call `CompressionCodecEncrypted::Configuration::instance().load()` instead.